### PR TITLE
Add MobileNetV3 models

### LINKED ISSUE #514: Include MobileNetv3 model

### PR DESCRIPTION
**Pull request by Rishit-dagli**
Ports the MobileNetV3 model introduced in [Searching for MobileNetV3](https://arxiv.org/pdf/1905.02244.pdf) (ICCV 2019) from `keras.applications`.

cc: @LukeWood 

Fixes #514

### CODE REVIEW

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:1]: add the KerasCV copyright notice to the top.

I'm surprised that the lint check isn't failing as it is missing - that is weird... will look into it.

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:21]: why does min_value need nonlocal?  What issue arises when you don't use this?

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:48]: same here, define multiply and HardSigmoid outside of the `apply()` block.

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:53]: Prefer fully spelled out names: SqueezeAndExcitationBlock.

Also note: https://github.com/keras-team/keras-cv/pull/505

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:58]: it is a little annoying, but you have to keep defining these outside of apply :(

I recommend using the SE block from https://github.com/keras-team/keras-cv/pull/505

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:81]: you can use the built in ReLU instead

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:86]: forgot return statement

**Review by LukeWood — CHANGES_REQUESTED** [review]
Some preliminary comments, why dont you take a stab at these and give it another pass.

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py:1]: Ah, yes, thanks for pointing that out.

> I'm surprised that the lint check isn't failing as it is missing - that is weird... will look into it.

Maybe because the workflows section, here on GitHub says:

> First-time contributors need a maintainer to approve running workflows

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: That does it and seems like a better solution than `nonlocal`

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Thanks for pointing this out

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Gotcha, sorry for missing this detail while writing the implementation 😓 and thanks for pointing it out at all places

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Yes, I'll also check if I have used improper names elsewhere in the code

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: I forgot to delete this part, was just doing some testing. All places in the code use `layers.ReLU` and not this. So I will just delete this

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Removed the new redundant ReLU layer altogether

**Comment by Rishit-dagli** [comment]
Thanks, @LukeWood for your insightful reviews, I have pushed all the changes you suggested. I will mark this PR as ready for further reviews and also start the work on writing out the docstrings for the model.

**Comment by Rishit-dagli** [comment]
@LukeWood I have added the docstrings as well

**Comment by LukeWood** [comment]
@Rishit-dagli please use the SqueezeAndExcite layer that has been merged into on master in place of the blocks here!

**Comment by Rishit-dagli** [comment]
The tests fail for this PR due to the commit https://github.com/keras-team/keras-cv/commit/0395a2a3355b78c267add6d7dc186a2260a0d4c2 , I have updated this PR accordingly.

**Comment by Rishit-dagli** [comment]
@LukeWood 
> @Rishit-dagli please use the SqueezeAndExcite layer that has been merged into on master in place of the blocks here!

I pushed commit https://github.com/keras-team/keras-cv/pull/526/commits/84f1a017eb670a20041b3d873b318a5226824bd4 which does this

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:122]: isn't depth just a function?  I don't see any use of name, or any Keras operations.  I think depth is just a utility function, and shouldn't be written as a closure.

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:200]: prefer fstrings:

f"expanded_conv_{block_id}"

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:203]: once you update Depth, change Depth()(...) => depth(...).

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:238]: do we need custom_object_scope?  Can we just pass `excite_activation=HardSigmoid()`?  or does that not work for some reason?

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:255]: is there ever a case where infilters != filters that is a valid case?

Comment (LukeWood) [reply keras_cv/models/mobilenet_v3.py:255]: if not, lets raise an error in that case

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:265]: what does ch stand for?

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:369]: any reason not to just rescale by 1/255?

**Review by LukeWood — CHANGES_REQUESTED** [review]
Some minor changes

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:19]: lowercase original

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:28]: lets just import `keras_cv.layers` and reference this as `keras_cv.layers.SqueezeAndExcite2D` in the code.

**Review by LukeWood — CHANGES_REQUESTED** [review]

**Comment by LukeWood** [comment]
@Rishit-dagli any updates here?

**Comment by Rishit-dagli** [comment]
Sorry, I am travelling for a conference which led to a delay; I will try to address the reviews on this and the MobileNet V2 PR as soon as possible

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py:120]: Yes, it's just a utility function

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Ah, yes

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Done, this everywhere we use `Depth()`

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py:236]: Yes this actually results in 

```
ValueError: Unknown activation function: apply. Please ensure this object is passed to the `custom_objects` argument. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.
```

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py:253]: Yes, infilters != filters would still be a valid case

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py:263]: channels: since this is the number of channels at the layer before the top

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Done

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Gotcha

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py:367]: The original `keras.applications` uses this rescaling in the preprocessing step, please see:

https://github.com/keras-team/keras/blob/v2.9.0/keras/applications/mobilenet_v2.py#L517

https://github.com/keras-team/keras/blob/07e13740fd181fc3ddec7d9a594d8a08666645f6/keras/applications/imagenet_utils.py#L191-L194

Thus, I thought to go ahead with this

**Comment by Rishit-dagli** [comment]
@LukeWood Thanks for the insightful reviews, I have incorporated them.

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:56]: please update the docstring to include the new parameters:

include_rescaling,
include_top, etc

Also please check the order of the arguments to match the constructors.

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Got you

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py:56]: Thanks

- New parameters
  - `include_top` is already present https://github.com/keras-team/keras-cv/blob/a314949a7d8cd8332f52175a542dc996b95f936a/keras_cv/models/mobilenet_v3.py#L72
  - fixed `include_rescaling` in docstrings
  - checked for all the other parameters in all functions
- The order of arguments already matches the constructors for all functions in this PR except `depth()` (which is now fixed), did I miss something?

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:494]: please update all constructors to match the API of DenseNet.

https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/densenet.py#L255

include_rescaling, include_top required.  IF include_top true, num_classes required.

Also update the underlying helper to match this API

**Review by LukeWood — CHANGES_REQUESTED** [review]
Good job and thanks for the PR, some API feedback.  Please update1

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Gotcha

**Comment by Rishit-dagli** [comment]
Thanks @LukeWood for the reviews, I have incorporated them ✅

File: keras_cv/models/models_test.py
Comment (LukeWood) [inline keras_cv/models/models_test.py:32]: please uncomment these before merging.

File: keras_cv/models/mobilenet_v3.py
Comment (LukeWood) [inline keras_cv/models/mobilenet_v3.py:261]: please remove defaults for include_rescaling and include_top.

File: keras_cv/models/models_test.py
Comment (Rishit-dagli) [reply keras_cv/models/models_test.py:32]: Ah, sorry, I didn't realize I pushed this. I did this while running the tests, to just test for MobileNet models.

File: keras_cv/models/mobilenet_v3.py
Comment (Rishit-dagli) [reply keras_cv/models/mobilenet_v3.py]: Gotcha

**Comment by Rishit-dagli** [comment]
Thanks @LukeWood for the reviews, I have incorporated them ✅

**Comment by Rishit-dagli** [comment]
@LukeWood The merge conflicts have been resolved

**Comment by Rishit-dagli** [comment]
@LukeWood I pushed a new commit, can you approve the test to run. The earlier test failed due to the ton of new changes to the tests merged since my last commit.

**Comment by LukeWood** [comment]
Hey @Rishit-dagli I am trying to run this over the finish line - hold off on committing for a bit!

**Comment by LukeWood** [comment]
Thank you @Rishit-dagli for your contribution!

**Pull request merged**
